src/extension/executors/task.ts

```diff
@@ -75,7 +75,8 @@ async function taskExecutor(
     //   return new ExperimentalTerminal(scriptFile.fsPath, { cwd, env })
     // })
   )
-  const isBackground = exec.cell.metadata?.['background'] === 'true'
+  const metadata = getMetadata(exec.cell)
+  const isBackground = metadata?.['background'] === 'true'
```

Comment:
Wouldn't this work too?

```suggestion
  const isBackground = exec.cell.metadata?.background === 'true'
```